File: src/application/decorator/interpreter/decorator_interpreter_factory.py
from application.decorator.interpreter.decorator_interpreter import DecoratorInterpreter
from application.decorator.strategies.decorator.decorator_helper_strategy import DecoratorHelperStrategy

class DecoratorInterpreterFactory:

    @staticmethod
    def create() -> DecoratorInterpreter :
        strategies = DecoratorHelperStrategy.create_instances()
        return DecoratorInterpreter(strategies)

File: angular-app/src/app/core/application/decorator/interpreter/decorator-interpreter-factory.ts
import { DecoratorHelperStrategy } from '../helpers';
import { DecoratorInterpreter } from './decorator-interpreter';

export class DecoratorInterpreterFactory {
  static create(): DecoratorInterpreter {
    const strategies = DecoratorHelperStrategy.createInstances();
    return new DecoratorInterpreter(strategies);
  }
}
